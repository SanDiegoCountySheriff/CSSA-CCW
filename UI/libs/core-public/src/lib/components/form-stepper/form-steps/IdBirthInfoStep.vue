<!-- eslint-disable @intlify/vue-i18n/no-raw-text -->
<template>
  <div>
    <v-form
      ref="form"
      v-model="valid"
    >
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Id Information') }}
      </v-subheader>
      <v-row>
        <v-col
          cols="12"
          lg="6"
        >
          <v-tooltip bottom>
            <template #activator="{ on, attrs }">
              <v-text-field
                dense
                outlined
                maxlength="25"
                counter
                persistent-hint
                v-model="completeApplication.idInfo.idNumber"
                :label="$t('Id number')"
                :hint="$t('PERMIT-HINT')"
                :rules="[v => !!v || $t('Id  number is required')]"
                v-bind="attrs"
                v-on="on"
              >
                <template #prepend>
                  <v-icon
                    x-small
                    color="error"
                  >
                    mdi-star
                  </v-icon>
                </template>
              </v-text-field>
            </template>
            {{
              $t(
                ' IMPORTANT! Must exactly match the information on your drivers license or id.'
              )
            }}
          </v-tooltip>
        </v-col>

        <v-col
          cols="12"
          lg="6"
        >
          <v-autocomplete
            outlined
            dense
            autocomplete="none"
            :items="states"
            :label="$t(' Issuing State')"
            :rules="[v => !!v || $t('Issuing state is required')]"
            v-model="completeApplication.idInfo.issuingState"
          >
            <template #prepend>
              <v-icon
                x-small
                color="error"
              >
                mdi-star
              </v-icon>
            </template>
          </v-autocomplete>
        </v-col>
      </v-row>

      <v-divider />
      <v-subheader class="sub-header font-weight-bold">
        {{ $t(' Date of birth') }}
      </v-subheader>

      <v-row>
        <v-col
          cols="12"
          lg="6"
        >
          <v-menu
            v-model="menu"
            :close-on-content-click="false"
            transition="scale-transition"
            offset-y
            min-width="auto"
          >
            <template #activator="{ on, attrs }">
              <v-text-field
                outlined
                dense
                readonly
                v-model="completeApplication.dob.birthDate"
                :label="$t('Date of Birth')"
                :rules="[
                  checkFor21,
                  v => !!v || $t('Date of birth is required'),
                ]"
                prepend-icon="mdi-calendar"
                v-bind="attrs"
                v-on="on"
              ></v-text-field>
            </template>
            <v-date-picker
              v-model="completeApplication.dob.birthDate"
              no-title
              scrollable
            >
            </v-date-picker>
          </v-menu>
        </v-col>

        <v-col
          cols="12"
          lg="6"
        >
          <v-text-field
            outlined
            dense
            maxlength="150"
            counter
            :label="$t('Birth city')"
            :rules="[v => !!v || $t('Birth city cannot be blank')]"
            v-model="completeApplication.dob.birthCity"
          >
            <template #prepend>
              <v-icon
                x-small
                color="error"
              >
                mdi-star
              </v-icon>
            </template>
          </v-text-field>

          <v-combobox
            outlined
            dense
            combobox="none"
            :items="countries"
            :label="$t('Birth country')"
            :rules="[v => !!v || $t('Birth country cannot be blank')]"
            v-model="completeApplication.dob.birthCountry"
          >
            <template #prepend>
              <v-icon
                x-small
                color="error"
              >
                mdi-star
              </v-icon>
            </template>
          </v-combobox>

          <v-autocomplete
            v-if="completeApplication.dob.birthCountry === 'United States'"
            outlined
            dense
            maxlength="150"
            counter
            autocomplete="none"
            :items="states"
            :label="$t('Birth state')"
            :rules="[v => !!v || $t('Birth state cannot be blank')]"
            v-model="completeApplication.dob.birthState"
          >
            <template #prepend>
              <v-icon
                x-small
                color="error"
              >
                mdi-star
              </v-icon>
            </template>
          </v-autocomplete>

          <v-text-field
            v-if="completeApplication.dob.birthCountry !== 'United States'"
            outlined
            dense
            maxlength="150"
            counter
            :label="$t('Birth region')"
            :rules="[v => !!v || $t('Birth region cannot be blank')]"
            v-model="completeApplication.dob.birthState"
          >
            <template #prepend>
              <v-icon
                x-small
                color="error"
              >
                mdi-star
              </v-icon>
            </template>
          </v-text-field>
        </v-col>
      </v-row>

      <v-divider />
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Citizenship Information') }}
      </v-subheader>

      <v-row>
        <v-col
          cols="12"
          lg="6"
          class="pl-5"
        >
          <v-radio-group
            :label="'Citizen'"
            :layout="'row'"
            v-model="completeApplication.citizenship.citizen"
          >
            <v-radio
              :color="$vuetify.theme.dark ? 'info' : 'primary'"
              :label="'Yes'"
              :value="true"
            />
            <v-radio
              :color="$vuetify.theme.dark ? 'info' : 'primary'"
              :label="'No'"
              :value="false"
            />
          </v-radio-group>
        </v-col>

        <v-col
          cols="12"
          lg="6"
        >
          <v-select
            outlined
            dense
            v-model="completeApplication.citizenship.militaryStatus"
            :items="items"
            :label="$t('Military Status')"
            :rules="[v => !!v || $t('Military Status is required')]"
          />

          <v-alert
            dense
            outlined
            type="warning"
            v-if="
              completeApplication.citizenship.militaryStatus === 'Discharged'
            "
          >
            {{ $t('discharged-disclaimer') }}
          </v-alert>
        </v-col>
      </v-row>

      <v-container
        fluid
        v-if="!completeApplication.citizenship.citizen"
      >
        <v-divider class="mb-3" />
        <v-subheader class="sub-header font-weight-bold">
          {{ $t('Immigrant Information') }}
        </v-subheader>
        <v-row class="ml-5">
          <v-col>
            <v-autocomplete
              autocomplete="none"
              outlined
              dense
              :items="countries"
              :label="$t('Country of Citizenship')"
              :rules="[v => !!v || $t('You must enter a country')]"
              v-model="
                completeApplication.immigrantInformation.countryOfCitizenship
              "
            >
              <template #prepend>
                <v-icon
                  x-small
                  color="error"
                >
                  mdi-star
                </v-icon>
              </template>
            </v-autocomplete>
            <v-radio-group
              :label="'Immigrant Alien'"
              v-model="completeApplication.immigrantInformation.immigrantAlien"
              row
            >
              <v-radio
                :color="$vuetify.theme.dark ? 'info' : 'primary'"
                :label="'Yes'"
                :value="true"
              />
              <v-radio
                :color="$vuetify.theme.dark ? 'info' : 'primary'"
                :label="'No'"
                :value="false"
              />
            </v-radio-group>
            <v-radio-group
              :label="'Non-Immigrant Alien'"
              row
              v-model="
                completeApplication.immigrantInformation.nonImmigrantAlien
              "
            >
              <v-radio
                :color="$vuetify.theme.dark ? 'info' : 'primary'"
                :label="'Yes'"
                :value="true"
              />
              <v-radio
                :color="$vuetify.theme.dark ? 'info' : 'primary'"
                :label="'No'"
                :value="false"
              />
            </v-radio-group>
          </v-col>
          <v-col
            cols="12"
            lg="6"
          >
            <v-combobox
              outlined
              dense
              combobox="none"
              :items="countries"
              :label="$t('Country of Birth')"
              :rules="[v => !!v || $t('You must enter a country')]"
              v-model="completeApplication.dob.birthCountry"
            >
              <template #prepend>
                <v-icon
                  x-small
                  color="error"
                >
                  mdi-star
                </v-icon>
              </template>
            </v-combobox>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
    <v-divider />
    <FormButtonContainer
      :valid="valid"
      :submitting="submited"
      @submit="handleSubmit"
      @save="saveMutation.mutate"
      @back="handlePreviousSection"
      @cancel="router.push('/')"
    />
    <v-snackbar
      :value="snackbar"
      :timeout="3000"
      outlined
      color="error"
    >
      {{ $t('Section update unsuccessful please try again.') }}
    </v-snackbar>
    <v-snackbar
      :value="formError"
      :timeout="3000"
      outlined
      color="error"
    >
      {{ $t('Military status has not been selected and is required') }}
    </v-snackbar>
  </div>
</template>

<script setup lang="ts">
import FormButtonContainer from '@shared-ui/components/containers/FormButtonContainer.vue';
import { TranslateResult } from 'vue-i18n';
import { i18n } from '@core-public/plugins';
import { ref } from 'vue';
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication';
import { useMutation } from '@tanstack/vue-query';
import { useRouter } from 'vue-router/composables';
import { countries, states } from '@shared-utils/lists/defaultConstants';

interface FormStepTwoProps {
  handleNextSection: () => void;
  handlePreviousSection: CallableFunction;
}

const props = withDefaults(defineProps<FormStepTwoProps>(), {
  handleNextSection: () => null,
});

const router = useRouter();

const items = ref(['Active', 'Reserve', 'Discharged', 'Retired', 'N/A']);
const snackbar = ref(false);
const valid = ref(false);
const menu = ref(false);
const submited = ref(false);
const formError = ref(false);

const completeApplicationStore = useCompleteApplicationStore();
const completeApplication =
  completeApplicationStore.completeApplication.application;

const updateMutation = useMutation({
  mutationFn: () => {
    completeApplication.immigrantInformation.countryOfBirth =
      completeApplication.dob.birthCountry;

    return completeApplicationStore.updateApplication();
  },
  onSuccess: () => {
    completeApplication.currentStep = 3;
    props.handleNextSection();
  },
  onError: () => {
    submited.value = false;
    valid.value = true;
    snackbar.value = true;
  },
});

const saveMutation = useMutation({
  mutationFn: () => {
    return completeApplicationStore.updateApplication();
  },
  onSuccess: () => {
    router.push('/');
  },
  onError: () => {
    submited.value = false;
    valid.value = true;
    snackbar.value = true;
  },
});

function handleSubmit() {
  // TODO: see about abstracting the if statements.
  if (!completeApplication.differentMailing) {
    completeApplication.mailingAddress.zip = '';
    completeApplication.mailingAddress.city = '';
    completeApplication.mailingAddress.state = '';
    completeApplication.mailingAddress.county = '';
    completeApplication.mailingAddress.country = '';
    completeApplication.mailingAddress.addressLine1 = '';
    completeApplication.mailingAddress.addressLine2 = '';
  }

  if (!completeApplication.differentSpouseAddress) {
    completeApplication.spouseAddressInformation.addressLine1 = '';
    completeApplication.spouseAddressInformation.addressLine2 = '';
    completeApplication.spouseAddressInformation.zip = '';
    completeApplication.spouseAddressInformation.country = '';
    completeApplication.spouseAddressInformation.county = '';
    completeApplication.spouseAddressInformation.city = '';
    completeApplication.spouseAddressInformation.state = '';
  }

  if (completeApplication.citizenship.citizen) {
    completeApplication.immigrantInformation.countryOfCitizenship = '';
    completeApplication.immigrantInformation.immigrantAlien = false;
    completeApplication.immigrantInformation.nonImmigrantAlien = false;
  }

  valid.value = false;
  submited.value = false;
  updateMutation.mutate();
}

function checkFor21(input: string): boolean | TranslateResult {
  const userDate = input;
  const targetDate = new Date(Date.now());
  const formatedDate = `${targetDate.getFullYear() - 21}-${
    targetDate.getMonth() + 1
  }-${targetDate.getDate()}`;

  return userDate <= formatedDate
    ? true
    : i18n.t('You must be 21 or older to apply for a CCW permit');
}
</script>
