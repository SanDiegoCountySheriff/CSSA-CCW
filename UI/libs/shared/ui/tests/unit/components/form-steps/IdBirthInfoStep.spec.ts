// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import IdBirthInfoStep from '@shared-ui/components/form-stepper/form-steps/IdBirthInfoStep.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('IdBirthInfoStep', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(IdBirthInfoStep, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
        handlePreviousSection: () => null,
      },
      data() {
        return {
          DOBInfo: {
            DOB: '',
          },
        };
      },
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render all the text inputs', () => {
    const textFields = wrapper.findAllComponents('.v-text-field');

    expect(textFields).toHaveLength(7);
  });
});
