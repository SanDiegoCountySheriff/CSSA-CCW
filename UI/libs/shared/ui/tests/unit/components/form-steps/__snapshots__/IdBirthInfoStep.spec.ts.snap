// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdBirthInfoStep should match the snapshot 1`] = `
<div>
  <form novalidate="novalidate" class="v-form">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Id Information
    </div>
    <div class="row">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">Id number</label><input aria-haspopup="true" aria-expanded="false" id="input-5" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-9" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;"> Issuing State</label><input autocomplete="none" id="input-9" type="text">
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Date of birth
    </div>
    <div class="row">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
          <div class="v-input__prepend-outer">
            <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate mdi mdi-calendar theme--light"></i></div>
          </div>
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-18" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-18" class="v-label theme--light" style="left: 0px; position: absolute;">Date of Birth</label><input role="button" aria-haspopup="true" aria-expanded="false" id="input-18" type="text" autocomplete="off">
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div class="v-menu">
          <!---->
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-24" class="v-label theme--light" style="left: 0px; position: absolute;">Birth city</label><input id="input-24" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-28" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-28" class="v-label theme--light" style="left: 0px; position: absolute;">Birth state</label><input autocomplete="none" id="input-28" type="text">
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-34" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-34" class="v-label theme--light" style="left: 0px; position: absolute;">Birth country</label><input autocomplete="none" id="input-34" type="text">
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Citizenship Information
    </div>
    <div class="row">
      <div class="pl-5 col-lg-6 col-12">
        <div layout="row" class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" aria-labelledby="input-42" class="v-input--radio-group__input">
                <legend id="input-42" class="v-label theme--light" style="left: 0px; position: relative;">Citizen</legend>
                <div class="v-radio theme--light v-item--active">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-43" role="radio" type="radio" name="radio-42" value="true">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-43" class="v-label theme--light" style="left: 0px; position: relative;">Yes</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-45" role="radio" type="radio" name="radio-42" value="false">
                    <div class="v-input--selection-controls__ripple"></div>
                  </div><label for="input-45" class="v-label theme--light" style="left: 0px; position: relative;">No</label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-49" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-49" class="v-label theme--light" style="left: 0px; position: absolute;">Military Status</label>
                <div class="v-select__selections"><input id="input-49" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div role="alert" class="v-alert v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined error--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--light error--text"></i>
            <div class="v-alert__content">
              Must select a status
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
    <!---->
  </form>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <div class="form-btn-container">
    <div class="form-btn-inner">
      <div><button type="button" disabled="disabled" class="mt-3 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"><span class="v-btn__content">
        Continue
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small accent mx-2"><span class="v-btn__content">
        Save and Exit
      </span></button></div>
      <div><button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small warning mx-2"><span class="v-btn__content">
        Go back
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error mr-2"><span class="v-btn__content">
        Cancel
      </span></button></div>
    </div>
  </div>
  <div class="v-snack v-snack--bottom">
    <transition-stub name="v-snack-transition">
      <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light error--text" style="display: none;">
        <div role="status" aria-live="polite" class="v-snack__content">
          Section update unsuccessful please try again.
        </div>
        <div class="v-snack__action "></div>
      </div>
    </transition-stub>
  </div>
</div>
`;
